core/computed/image-records.js

```diff
@@ -39,12 +38,13 @@ class ImageRecords {
 
     for (const element of data.ImageElements) {
       const networkRecord = indexedNetworkRecords[element.src];
-      const mimeType = networkRecord?.mimeType;
+      // Ignore if we aren't sure this is really an image.
+      // TODO: we should add something in tracing to get this straight from blink.
+      if (!networkRecord) continue;
 
-      // Don't change the guessed mime type if no mime type was found.
       imageRecords.push({
         ...element,
-        mimeType: mimeType ? mimeType : URL.guessMimeType(element.src),
+        mimeType: networkRecord.mimeType,
```

Comment:
brendan said:

> This might be going too aggressive, though :/ About 1.6% of sites in HTTP Archive (July 2022) currently have an image LCP which would no longer be detected after this. It is catching a bunch of accidental css files (at least going by filenames), but there also just seem to be a ton of people serving images as `application/images`, `application/png`, or good old `application/xml`.

and then I was like... "i thought we were gonna used `resourceType` but then see https://github.com/GoogleChrome/lighthouse/issues/13338#issuecomment-1215969640 where you showed an HTML page getting described as `resourceType: Image`


i tried to reproduce this locally and failed. but then tried `<img src="page.html">`. Yeah very interesting. 

These resourceTypes are [defined in the blink loader](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/platform/loader/fetch/resource.h;l=76-94;drc=9f5d860e6655962757102174e237da66015f15ee) and set (for non-scripts) [here](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/core/inspector/inspector_network_agent.cc;l=1363-1377;drc=9f5d860e6655962757102174e237da66015f15ee)…

I guess this type is still in the "loader" before it's determined that it's an invalid asset for the intended use.

Ah well. So you're right about resourceType not being useful for this situation.  

humph.